### ISSUE TITLE
Regulate temperature in room(overfitting, continuous action space) [question]

### ISSUE THREAD

**Issue by andrehoejmark** [hill-a/stable-baselines#277]
Hi, i'm trying to regulate temperature in a room with reinforcement learning and got some problems with continuous actions and over-fitting.

The model has only 1 input called power with values in range [0, 10000] and by changing power the model will show the next temperature 10 minutes into the future.

### Example
```
Target temperature 22
Current Temperature 19

step(power=5000) --> next_temperature: 19.2
step(power=400) --> next_temperature: 18.8
```

Reward function
`reward = -(self.next_temperature - self.target_temperature )**2`

Action space
`self.action_space = spaces.Box(low=0, high=10000, shape=(1,), dtype=np.float32) `

Observation Space
`self.observation_space = spaces.Box(low=0, high=70, shape=(1,), dtype=np.float32)`

What I've done so far is creating a continuous action space in range [0, 10000] but it seems tested actions to see which ones are good are only tested in range [0, 3] therefore i multiplied the tested power with 3000 to get in range [0, 10000].

**Question 1 - continuous action space**
Why does it not give power in range [0, 10000] i suppose it's normalized somehow?

**Question 2 - overfitting** 
I trained the model with target temperature 22 and when testing different learning rates & time steps it eventually gave quite good results. But when I tested the model with a different target temperature ex 20 the model still predicted power values that moved the model to 22 degrees even if the rewards could become lower.

**Question 3**
Got any tips or references for using stable baselines to regulate temperature in a room or anything similar.

**Comment by andrehoejmark** [comment]
https://github.com/andrehoejmark/Tempearture-control/

**Comment by araffin** [comment]
Hello,

> Question 1 - continuous action space
> Why does it not give power in range [0, 10000] i suppose it's normalized somehow?

For your first question, yes you should normalize (see [here](https://twitter.com/araffin2/status/1111983313676312576) for a little longer explanation). I assume you are using A2C or a derivative (PPO, ...), with SAC or DDPG you won't have that issue (but you would need a symmetric action space).
You should also normalize the observation space.

I think I will add at some point a section on tips for working with RL.
![normalization](https://pbs.twimg.com/media/D26OOvyXgAEV9Pc.jpg:large)

>Question 2 - overfitting
> Question 3
> Got any tips or references for using stable baselines to regulate temperature in a room or anything similar.

Well this behavior is not really surprising, you should train it in different conditions.

But in fact, your problem looks like something suited for using [Hindsight Experience Replay](https://arxiv.org/abs/1707.01495) (I'm currently working on that in #273 ) where you have explicit goal. This would allow to generalize to different scenarios.

For using stable-baselines in general, I would recommend you to take a look at the [rl zoo](https://github.com/araffin/rl-baselines-zoo) for the best practices. Otherwise, I am not aware of any related work.
